### PR TITLE
GUARD-280 GUARD-305 release

### PR DESCRIPTION
**Pull request by maxim-skuvault**

### CODE REVIEW

File: src/BigCommerceAccess/BigCommerceProductsServiceV3.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+					this._webRequestServices.PutData( BigCommerceCommand.UpdateProductsV3, endpoint, jsonContent, marker ) );
+				this.CreateApiDelay( limit ).Wait(); //API requirement
+			}
+		}
+
+		public async Task UpdateProductsAsync( List< BigCommerceProduct > products, CancellationToken token )
+		{
+			var marker = this.GetMarker();
+
+			await products.DoInBatchAsync( 20, async product =>
```
Comment (slav) [inline src/BigCommerceAccess/BigCommerceProductsServiceV3.cs]: I would use 5 here.

File: src/BigCommerceAccess/BigCommerceProductsServiceV2OAuth.cs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+					this._webRequestServices.PutData( BigCommerceCommand.UpdateProductV2_OAuth, endpoint, jsonContent, marker ) );
+				this.CreateApiDelay( limit ).Wait(); //API requirement
+			}
+		}
+
+		public async Task UpdateProductOptionsAsync( List< BigCommerceProductOption > productOptions, CancellationToken token )
+		{
+			var marker = this.GetMarker();
+
+			await productOptions.DoInBatchAsync( 20, async option =>
```
Comment (slav) [inline src/BigCommerceAccess/BigCommerceProductsServiceV2OAuth.cs]: I would use 5 here

File: src/BigCommerceAccess/BigCommerceProductsServiceV2OAuth.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+					this._webRequestServices.PutData( BigCommerceCommand.UpdateProductV2_OAuth, endpoint, jsonContent, marker ) );
+				this.CreateApiDelay( limit ).Wait(); //API requirement
+			}
+		}
+
+		public async Task UpdateProductsAsync( List< BigCommerceProduct > products, CancellationToken token )
+		{
+			var marker = this.GetMarker();
+
+			await products.DoInBatchAsync( 20, async product =>
```
Comment (slav) [inline src/BigCommerceAccess/BigCommerceProductsServiceV2OAuth.cs]: I would use 5 here

File: src/BigCommerceAccess/BigCommerceProductsServiceV3.cs [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+					this._webRequestServices.PutData( BigCommerceCommand.UpdateProductsV3, endpoint, jsonContent, marker ) );
+				this.CreateApiDelay( limit ).Wait(); //API requirement
+			}
+		}
+
+		public async Task UpdateProductOptionsAsync( List< BigCommerceProductOption > productOptions, CancellationToken token )
+		{
+			var marker = this.GetMarker();
+
+			await productOptions.DoInBatchAsync( 20, async productOption =>
```
Comment (slav) [inline src/BigCommerceAccess/BigCommerceProductsServiceV3.cs]: Use 5 here

**Comment by slav** [comment]
It seems in some places number of threads for batch operations is MaxThreadsCount and in other it's hardcoded. I'd recommend switching to using MaxThreadsCount, which also should be lowered to 5 from 10.

File: src/BigCommerceAccess/BigCommerceProductsServiceV3.cs (hunk 1 above)
Comment (maxim-skuvault) [reply src/BigCommerceAccess/BigCommerceProductsServiceV3.cs]: Changed here and everywhere else in the solution to use MaxThreadsCount, which I decreased from 10 to 5

**Pull request merged**